src/EventStore.ClientAPI/NoCompatibilityMode.cs

```diff
@@ -0,0 +1,20 @@
+namespace EventStore.ClientAPI {
+	/// <summary>
+	/// Disables all compatibility mode.
+	/// </summary>
+	public class NoCompatibilityMode : ICompatibilityMode {
+		/// <summary>
+		/// Is EventStoreDB Auto compatibility mode enabled.
+		/// </summary>
+		public bool IsAutoCompatibilityModeEnabled() {
+			return false;
+		}
+
+		/// <summary>
+		/// Is EventStoreDB Version 5 compatibility mode enabled.
+		/// </summary>
+		public bool IsVersion5CompatibilityModeEnabled() {
+			return true;
```

Comment:
This is confusing.  `NoCompatibilityMode.IsVersion5CompatibilityModeEnabled() == true`.

Perhaps we should just rename this class to `Version5CompatibilityMode` and use that if the compatibility mode is not explicitly set.  Same behaviour as now, just less confusing.

Reply:
Agree, it makes more sense.